src/util/prolog/prologSession.ts

```diff
@@ -17,11 +18,18 @@ export function importFile(relativePath: string): PrologSession {
 }
 
 function resolveImports(program: string, defaultPath: string, currentFileName: string = "main"): string {
-		const dependencyGraph: Set<string[]> = getDependencyGraph(program, defaultPath, currentFileName);
-		const importList: string[] = kahnAlgorithm(dependencyGraph);
+		const dependencyGraph: Set<Edge> = getDependencyGraph(program, defaultPath, currentFileName);
+		const importList: string[] | Set<Edge> = getTopologicalOrder(dependencyGraph);
+
+		// Fetching an error; see in getTopologicalOrder.ts
+		if (importList instanceof Set) {
+			console.log("Residual Graph: " + importList.toString);
+			throw new Error("Cyclic Import. See the console for more information.");
```

Comment:
Please throw this error from within `getTopologicalOrder`. Getting the residual graph as part of an error message is good enough, I don't think any code could reasonably "handle" these cases.